src/objects/SimpleWaterMesh.ts

```diff
@@ -0,0 +1,301 @@
+            float timeOffset = time * ripplesSpeed;
+            uv = (uv - 0.5) * (1.0 / ripplesScale);
+            vec2 uv0 = (uv/103.0)+vec2(timeOffset/17.0, timeOffset/29.0);
+            vec2 uv1 = uv/107.0-vec2(timeOffset/-19.0, timeOffset/31.0);
+            vec2 uv2 = uv/vec2(897.0, 983.0)+vec2(timeOffset/101.0, timeOffset/97.0);
+            vec2 uv3 = uv/vec2(991.0, 877.0)-vec2(timeOffset/109.0, timeOffset/-113.0);
+            vec4 noise = (texture2D(normalMap, uv0)) +
+                         (texture2D(normalMap, uv1)) +
+                         (texture2D(normalMap, uv2)) +
+                         (texture2D(normalMap, uv3));
+            return noise / 4.0;
+          }
+        `
+      );
+
+      // https://github.com/mrdoob/three.js/blob/dev/src/renderers/shaders/ShaderChunk/normalmap_pars_fragment.glsl.js#L20
+      shader.fragmentShader = shader.fragmentShader.replace(
+        "#include <normal_fragment_maps>",
+        `
+            // Workaround for Adreno 3XX dFd*( vec3 ) bug. See #9988
+  
+            vec3 eye_pos = -vViewPosition;
+            vec3 q0 = vec3( dFdx( eye_pos.x ), dFdx( eye_pos.y ), dFdx( eye_pos.z ) );
+            vec3 q1 = vec3( dFdy( eye_pos.x ), dFdy( eye_pos.y ), dFdy( eye_pos.z ) );
+            vec2 st0 = dFdx( vUv.st );
+            vec2 st1 = dFdy( vUv.st );
+  
+            float scale = sign( st1.t * st0.s - st0.t * st1.s ); // we do not care about the magnitude
+  
+            vec3 S = normalize( ( q0 * st1.t - q1 * st0.t ) * scale );
+            vec3 T = normalize( ( - q0 * st1.s + q1 * st0.s ) * scale );
+            vec3 N = normalize( normal );
+            mat3 tsn = mat3( S, T, N );
+  
+            vec3 mapN = getNoise(vWPosition.xz).xyz * 2.0 - 1.0;
+  
+            mapN.xy *= normalScale;
+            mapN.xy *= ( float( gl_FrontFacing ) * 2.0 - 1.0 );
+  
+            normal = normalize( tsn * mapN );
+          `
+      );
+    };
+
+    super(geometry, material);
+
+    this.resolution = resolution;
+    this.lowQuality = lowQuality;
+    this.waterMaterial = material;
+    this.waterUniforms = waterUniforms;
+
+    if (this.lowQuality) {
+      if (this.material instanceof MeshPhongMaterial) {
+        this.material.specular.set(0xffffff);
+      }
+    } else {
+      this.receiveShadow = true;
+    }
+
+    const buffer: BufferAttribute | InterleavedBufferAttribute = this.geometry.attributes.position;
+    if (buffer instanceof BufferAttribute) {
+      (buffer as BufferAttribute).setUsage(DynamicDrawUsage);
+    }
+
+    this.octaves = [
+      new Octave(new Vector2(0.5, 0.5), new Vector2(1, 1), 0.01, true),
+      new Octave(new Vector2(0.05, 6), new Vector2(1, 20), 0.1, false)
+    ];
+
+    this.simplex = new SimplexNoise();
+  }
+
+  get opacity(): number {
+    return this.waterMaterial.opacity;
+  }
+
+  set opacity(value: number) {
+    this.waterMaterial.opacity = value;
+    if (value !== 1) {
+      this.waterMaterial.transparent = true;
+    }
+  }
```

Comment:
Is it intentional that `.transparent` won't be `false` even if `.opacity` is 1?

Reply:
Yeah, that's true, it seems to me that we should always set...

Reply:
We may fix it or add a comment in another PR because the main purpose of this PR is migration, not fixing a bug.

Reply:
I don't know for sure but it does seem like this is supposed to say something like
```js
this.waterMaterial.transparent = value !== 1;
```
And please _do_ fix bugs like these as we migrate things because it's a great opportunity to make minor improvements.

Reply:
We do. And separating bug fix PR from migration PR is good to track the change later.

Reply:
Yep, fixing bugs like this is absolutely something we should do as we migrate these. Also find opportunities to simplify by dropping features we wrote code for but never exposed (check spoke and blender addon), etc. Re separate PRs, use best judgment. I think its ok to combine fixing with migrating if the migration PR isn't very complicated. Just make sure to note the change in the PR notes though, so the reviewer knows it was intentionally changed.

Reply:
To clarify, I didn't say we shouldn't fix bugs when migrating. We should fix problems when we find. The first reason why I suggested to make another PR is we didn't get quick response from you two. There are many historical and unclear code in the old Hubs Client code that new devs couldn't determine whether intentional or not. I didn't think it's good to block this PR until we get the comment from you.